### PR TITLE
Adds the ol' banana in the tailpipe

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
You can now stuff a banana in the tailpipe of a secway to prevent it from being able to chase you. The driver of the secway won't get a message that it's happening, but if you don't notice the guy walking up to your ride with a banana and standing around next to it, then you aren't a first team cop.

also i dedicate this PR to my mom because she loves beverly hill cop and especially the scene this is from. she was so happy to get to show me this movie, and even if she'll never see this stupid meme spessmen PR, i still wanna rep her. love u mom

## Why It's Good For The Game
Funny bit of flavor, actually fits the SS13 world shockingly well.

## Changelog
:cl: Ryll/Shaps
add: You can now sabotage secways by stuffing a banana in their tailpipe. 
/:cl:

### CODE REVIEW

**Comment by RaveRadbury** [comment]
TIL that secways have exhaust pipes.

**Comment by Ryll-Ryll** [comment]
don't worry about it man

**Comment by Shadowflame909** [comment]
> 
> 
> TIL that secways have exhaust pipes.

The fuel is assistant tears

**Comment by gitbirb** [comment]
this is great, it could only get better if it made secways explode after a bit

**Comment by ExcessiveUseOfCobblestone** [comment]
if you're putting the var on the obj instead of just using it locally you should probably name it something more obvious

**Comment by Ryll-Ryll** [comment]
please, the comment about cleaning eddie murphy out of the exhaust is probably the funniest code joke i've ever made, i beg of you

**Comment by PKPenguin321** [comment]
Doesn't this bit usually use a potato

**Comment by Ryll-Ryll** [comment]
https://www.youtube.com/watch?v=6y-pdLyZPJ8

File: code/modules/vehicles/secway.dm [hunk 1]
```diff
@@ -28,6 +31,14 @@
 	smoke.start()
 
 /obj/vehicle/ridden/secway/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/reagent_containers/food/snacks/grown/banana))
+		// ignore the occupants because they're presumably too distracted to notice the guy stuffing fruit into their vehicle's exhaust. do segways have exhausts? they do now!
+		user.visible_message("<span class='warning'>[user] begins stuffing [W] into [src]'s tailpipe.</span>", "<span class='warning'>You begin stuffing [W] into [src]'s tailpipe...</span>", ignored_mobs = occupants)
+		if(do_after(user, 30, target = src))
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: ` 		if(do_after(user, 30, TRUE, src))`

Comment (ShizCalev) [reply code/modules/vehicles/secway.dm]: otherwise if the banana leaves your hand, you'll still put it in.

File: code/modules/vehicles/secway.dm [hunk 2]
```diff
@@ -7,6 +7,9 @@
 	armor = list("melee" = 10, "bullet" = 0, "laser" = 10, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 60, "acid" = 60)
 	key_type = /obj/item/key/security
 	integrity_failure = 0.5
+	var/last_rev
+
+	var/obj/item/reagent_containers/food/snacks/grown/banana/eddie_murphy
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: please make this var less vague, and autodoc what it does with `/// if it has a banana in the pipe, it does x`

File: code/modules/vehicles/secway.dm [hunk 3]
```diff
@@ -38,6 +49,30 @@
 		return TRUE
 	return ..()
 
+/obj/vehicle/ridden/secway/attack_hand(mob/living/user)
+	if(eddie_murphy)                                                       // v lol
+		user.visible_message("<span class='warning'>[user] begins cleaning [eddie_murphy] out of [src].</span>", "<span class='warning'>You begin cleaning [eddie_murphy] out of [src]...</span>")
+		if(do_after(user, 60, target = src))
+			user.visible_message("<span class='warning'>[user] cleans [eddie_murphy] out of [src].</span>", "<span class='warning'>You manage to get [eddie_murphy] out of [src].</span>")
+			eddie_murphy.forceMove(get_turf(src))
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: drop_location()

File: code/modules/vehicles/secway.dm [hunk 4]
```diff
@@ -38,6 +49,30 @@
...
+			eddie_murphy = null
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: return at the end

File: code/modules/vehicles/secway.dm [hunk 5]
```diff
@@ -28,6 +31,14 @@
 	smoke.start()
 
 /obj/vehicle/ridden/secway/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/reagent_containers/food/snacks/grown/banana))
+		// ignore the occupants because they're presumably too distracted to notice the guy stuffing fruit into their vehicle's exhaust. do segways have exhausts? they do now!
+		user.visible_message("<span class='warning'>[user] begins stuffing [W] into [src]'s tailpipe.</span>", "<span class='warning'>You begin stuffing [W] into [src]'s tailpipe...</span>", ignored_mobs = occupants)
+		if(do_after(user, 30, target = src))
+			user.visible_message("<span class='warning'>[user] stuffs [W] into [src]'s tailpipe.</span>", "<span class='warning'>You stuff [W] into [src]'s tailpipe.</span>", ignored_mobs = occupants)
+			eddie_murphy = W
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: didn't check if it's nodrop

File: code/modules/vehicles/secway.dm [hunk 6]
```diff
@@ -7,6 +7,9 @@
 	armor = list("melee" = 10, "bullet" = 0, "laser" = 10, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 60, "acid" = 60)
 	key_type = /obj/item/key/security
 	integrity_failure = 0.5
+	var/last_rev
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: what's this do. comment it please.

File: code/modules/vehicles/secway.dm [hunk 7]
```diff
@@ -28,6 +31,14 @@
 	smoke.start()
 
 /obj/vehicle/ridden/secway/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/reagent_containers/food/snacks/grown/banana))
+		// ignore the occupants because they're presumably too distracted to notice the guy stuffing fruit into their vehicle's exhaust. do segways have exhausts? they do now!
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: check if it's nodrop via `!user.transferItemToLoc(W, src)` before you do any of the stuff.

**Comment by JMoldy** [comment]
Wouldn't it make more sense to do this to clown cars? Aren't secways electric vehicles?

**Comment by Ryll-Ryll** [comment]
See the first and second comments on this PR for more information

File: code/modules/vehicles/secway.dm (hunk 2 above)
Comment (optimumtact) [reply code/modules/vehicles/secway.dm]: it's fine, just needs a comment

File: code/modules/vehicles/secway.dm [hunk 8]
```diff
@@ -47,23 +43,35 @@
...
 					to_chat(user, "<span class='notice'>It looks to be fully repaired now.</span>")
 		return TRUE
+
+	if(istype(W, /obj/item/reagent_containers/food/snacks/grown/banana))
+		if(!user.transferItemToLoc(W, src))
+			return
+		// ignore the occupants because they're presumably too distracted to notice the guy stuffing fruit into their vehicle's exhaust. do segways have exhausts? they do now!
+		user.visible_message("<span class='warning'>[user] begins stuffing [W] into [src]'s tailpipe.</span>", "<span class='warning'>You begin stuffing [W] into [src]'s tailpipe...</span>", ignored_mobs = occupants)
+		if(do_after(user, 30, TRUE, target = src))
+			user.visible_message("<span class='warning'>[user] stuffs [W] into [src]'s tailpipe.</span>", "<span class='warning'>You stuff [W] into [src]'s tailpipe.</span>", ignored_mobs = occupants)
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: actually rethinking this, move the transferitemtoloc() above here instead, it'll handle the forceMove() as well.

File: code/modules/vehicles/secway.dm [hunk 9]
```diff
@@ -36,8 +43,44 @@
 				if(obj_integrity == max_integrity)
 					to_chat(user, "<span class='notice'>It looks to be fully repaired now.</span>")
 		return TRUE
+
+	if(istype(W, /obj/item/reagent_containers/food/snacks/grown/banana))
+		if(!user.transferItemToLoc(W, src))
+			return
+		// ignore the occupants because they're presumably too distracted to notice the guy stuffing fruit into their vehicle's exhaust. do segways have exhausts? they do now!
+		user.visible_message("<span class='warning'>[user] begins stuffing [W] into [src]'s tailpipe.</span>", "<span class='warning'>You begin stuffing [W] into [src]'s tailpipe...</span>", ignored_mobs = occupants)
+		if(do_after(user, 30, TRUE, target = src))
```
Comment (ShizCalev) [inline code/modules/vehicles/secway.dm]: change target = src to just src.

**Review by kriskog — APPROVED** [review]

**Pull request merged**